### DISCUSSION TITLE
Create strongswan IKEv2 VPN without domain name (just with ip adress)

### DISCUSSION THREAD

**Discussion by mparkhomenko** [strongswan/strongswan#1294] [Q&A]
Hello everyone! I want to create an IKEv2 VPN server without a domain name and with EAP (username: password) auth. I read this [article](https://www.howtoforge.com/tutorial/how-to-setup-ikev2-vpn-using-strongswan-and-letsencrypt-on-centos-7/) but in this case, we need to use a domain name for generating certificates, but when I want to use the same command I see that letsencrypt says that it doesn't work with IP (only with domain names). I didn't find any solution...

How I can change settings from an article which I paste above to do this? Or maybe you can send me an article for reading. Thank you!

**Comment by seldfv** [comment]
strongswan has a tool to create certificates : https://docs.strongswan.org/docs/5.9/pki/pkiQuickstart.html
There is another tutorial there that uses the tool and that shows how to use your IP address instead of a domain name : https://www.digitalocean.com/community/tutorials/how-to-set-up-an-ikev2-vpn-server-with-strongswan-on-ubuntu-20-04

**Reply by mparkhomenko** [reply]
Thank you! But in this tutorial, I need to install a certificate on a local computer "Once you have the ca-cert.pem file downloaded to your computer, you can set up the connection to the VPN.". I want to do the same from an article, but without installation certificate on a local machine

**Reply by tobiasbrunner** [reply]
That's how it works. Use a domain name and Let's Encrypt (what's the problem with that anyway?), or either find a commercial CA that is already trusted by your clients and that issues you a certificate for an IP address (you'll have to prove that you own that address), or, as @seldfv described, create your own PKI for which you'll have to install the root certificate on your clients.
